src/communication/domain_search_queue_policy.hpp

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#ifndef NMC_USE_DOMAIN_SEARCH_QUEUE
+#error "domain_search_queue_policy.hpp should only be compiled in a NMC_USE_DOMAIN_SEARCH_QUEUE build"
```

Comment:
I don't think you would --- something is really f*** if the error is called, and you'd need to hunt from CMakeLists down to figure it out. I don't think there's a better solution, other than adding a pointer to src/communications/communicator.hpp